mantidproject/mantid
# Fix common widgets tests when Qt5 libraries are present

**Description of work**

Th plugin library loader has been extended to allow the excluded set of libraries to be defined externally. This allows the `InterfaceManager` at the Qt layer to exclude plugins that are not built for the version of Qt it is compiled against. The properties file has also been extended to allow exclusions of libraries for just the framework so prevent any Qt libraries being loaded when the framework is used alone.

**To test:**

* build the project with `-DENABLE_WORKBENCH=ON`.
* run the common widgets tests: `ctest -R MantidQtWidgetsCommonTest`
* there should be no segfaults

*No associated issue*

**Release Notes** 

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
